include/openssl/pem.h

```diff
@@ -187,6 +188,120 @@ extern "C" {
         IMPLEMENT_PEM_read(name, type, str, asn1) \
         IMPLEMENT_PEM_write_cb(name, type, str, asn1)
 
+/* Alternatives using provided serializers */
```

Comment:
That can be said about *all* IMPLEMENT macros.  They could absolutely be moved out of the way, but in that case they should *all* be moved out of the way.  Another PR.

Reply:
I (strongly) disagree. Just because we did something wrong before, it doesn't mean we should do new stuff wrongly. There is nothing preventing us from putting this stuff in a private header, whilst leaving existing IMPLEMENT macros where they are.

Reply:
Seems this is still outstanding

Reply:
I disagree with spreading things like this over several headers.  It makes this sort of stuff harder to follow.  However, I'll do that for this PR, with the intent to follow up with another PR that moves *all* IMPLEMENT macros.